### PR TITLE
8252830: Correct missing javadoc comments in java.rmi module

### PR DESCRIPTION
**Pull request by RogerRiggs**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8252830](https://bugs.openjdk.java.net/browse/JDK-8252830): Correct missing javadoc comments in java.rmi module

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Stuart Marks](https://openjdk.java.net/census#smarks) (@stuart-marks - **Reviewer**)
 * [Chris Hegarty](https://openjdk.java.net/census#chegar) (@ChrisHegarty - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/79/head:pull/79`
`$ git checkout pull/79`

### CODE REVIEW

**Comment by RogerRiggs** [comment]
Add the missing comments where required by javadoc -Xdoclint to document public and serializable classes, methods, and fields.

File: src/java.rmi/share/classes/java/rmi/activation/ActivationGroupDesc.java [hunk 1]
```diff
@@ -308,11 +308,15 @@ public int hashCode()
...
          * the specified object input stream, and if <code>options</code>
          * is <code>null</code>, then <code>options</code> is set to a
          * zero-length array of <code>String</code>.
+         *
+         * @param  in the {@code ObjectInputStream} from which data is read
+         * @throws IOException if an I/O error occurs
+         * @throws ClassNotFoundException if a serialized class cannot be loaded
+         *
          */
         private void readObject(ObjectInputStream in)
```
Comment (dfuch) [inline src/java.rmi/share/classes/java/rmi/activation/ActivationGroupDesc.java:321]: Hi Roger, you could take this opportunity to add the @serial annotation to this method (here and in the other files you modified as well.

File: src/java.rmi/share/classes/java/rmi/activation/ActivationID.java [hunk 2]
```diff
@@ -228,6 +228,9 @@ public boolean equals(Object obj) {
      * java.rmi.server.RemoteObject RemoteObject}
      * <code>writeObject</code> method <b>serialData</b>
      * specification.
+     *
+     * @param  out the {@code ObjectOutputStream} to which data is written
+     * @throws IOException if an I/O error occurs
      **/
     private void writeObject(ObjectOutputStream out)
```
Comment (dfuch) [inline src/java.rmi/share/classes/java/rmi/activation/ActivationID.java:235]: Same comment as above...

**Review by dfuch — CHANGES_REQUESTED** [review]

**Review by LanceAndersen — APPROVED** [review]
Hi Roger,

The combined changes to both commits look good

File: src/java.rmi/share/classes/java/rmi/activation/ActivationID.java [hunk 3]
```diff
@@ -228,6 +229,9 @@ public boolean equals(Object obj) {
...
```
Comment (dfuch) [inline src/java.rmi/share/classes/java/rmi/activation/ActivationID.java:236]: one `@Serial` still missing here?

**Review by dfuch — APPROVED** [review]
Otherwise looks good.

**Review by LanceAndersen — APPROVED** [review]

**Review by dfuch — APPROVED** [review]

File: src/java.rmi/share/classes/java/rmi/server/RemoteObject.java [hunk 4]
```diff
@@ -357,7 +359,11 @@ public String toString() {
      * <code>"UnicastServerRef2"</code>, no data is written by the
      * <code>writeExternal</code> method or read by the
      * <code>readExternal</code> method.
+     *
+     * @param  out the {@code ObjectOutputStream} to which data is written
+     * @throws IOException if an I/O error occurs
      */
+    @java.io.Serial
     private void writeObject(java.io.ObjectOutputStream out)
         throws java.io.IOException
```
Comment (stuart-marks) [inline src/java.rmi/share/classes/java/rmi/server/RemoteObject.java:368]: It's a little odd to see an import of java.io.IOException (which is used by the `@throws` tag) but not to have it used here. There's another occurrence down below. These files have a mixture of imports and fully qualified names, which I find irritating, which might be nice to clean up. But you might find beyond the scope of this change. So, clean up as much of this as you like, or none if you prefer, no big deal.

**Review by stuart-marks — APPROVED** [review]

**Review by ChrisHegarty — APPROVED** [review]

**Comment by RogerRiggs** [comment]
Hi Stuart,

I'm going to leave this for later, its a bit out of scope and every change
prompts the people who have already reviewed it to need to review it again.

Thanks, Roger

**Pull request closed without merging**